cmd/gpu-kubelet-plugin/driver.go

```diff
@@ -30,12 +31,38 @@ import (
 )
 
 type driver struct {
-	sync.Mutex
 	client       coreclientset.Interface
 	pluginhelper *kubeletplugin.Helper
 	state        *DeviceState
 }
 
+// Use this function to protect the work in nodePrepareResource() and
+// nodeUnprepareResource() under a lock. A file-based lock was chosen because
+// more than one driver pod may be running on a single node, but at most one
+// such function must execute at any given time.
+func acquirePrepUnprepLock() (func(), error) {
```

Comment:
(not yet in sync with CD code -- and yes, we really should deduplicate :))

Reply:
We should also build an abstraction around with this with a simple Lock() / Unlock() interface (maybe that already exists in go stdlib somewhere) and a NewFlock(path) function that returns a type that implements that interface.

The we can continue to do the following:
```
	d.Lock()
	defer d.Unlock()
```

But the lock that's backing it is an flock.

Reply:
note from in sync meet: put this in `pkg`.